src/name/verify.rs

```diff
@@ -272,6 +277,33 @@ fn iterate_names(
     }
 }
 
+#[cfg(feature = "alloc")]
+pub fn list_cert_dns_names<'names>(
+    cert: &'names crate::EndEntityCert<'names>,
+) -> Result<Vec<GeneralDnsNameRef<'names>>, Error> {
```

Comment:
How hard would it be to rewire this to return an `impl Iterator` instead?

Reply:
it certainly could be rewritten to return `impl Iterator` --- I wanted to start by coping the original diff from upstream exactly, but would be happy to revise.